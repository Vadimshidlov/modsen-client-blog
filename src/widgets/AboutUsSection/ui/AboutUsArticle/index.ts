export { AboutUsArticle } from '@/widgets/AboutUsSection/ui/AboutUsArticle/AboutUsArticle';
