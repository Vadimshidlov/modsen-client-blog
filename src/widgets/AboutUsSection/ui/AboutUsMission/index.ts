export { AboutUsMission } from '@/widgets/AboutUsSection/ui/AboutUsMission/AboutUsMission';
