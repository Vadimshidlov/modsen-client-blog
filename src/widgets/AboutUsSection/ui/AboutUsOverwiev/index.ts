export { AboutUsOverwiev } from '@/widgets/AboutUsSection/ui/AboutUsOverwiev/AboutUsOverwiev';
