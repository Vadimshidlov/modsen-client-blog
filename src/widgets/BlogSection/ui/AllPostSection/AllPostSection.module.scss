.allPostSection {
  display: flex;
  flex-direction: column;
  align-items: start;
  margin-bottom: 68px;

  .sectionTitle {
    padding-top: 64px;
    width: 100%;
    @include headingOne;
    margin-bottom: 64px;
    padding-bottom: 32px;
    border-bottom: 1px solid $grey;

    @media (max-width: $tablet) {
      @include headingTwo;
      padding-top: 32px;
      margin-bottom: 32px;
    }

    @media (max-width: $mobile) {
      @include headingThree;
      padding-top: 16px;
      margin-bottom: 16px;
    }
  }

  .postsContainer {
    display: flex;
    flex-direction: column;
    row-gap: 64px;
    margin-bottom: 64px;

    @media (max-width: $tablet) {
      row-gap: 32px;
    }
  }

  .paginationController {
    align-self: center;
    display: flex;
    column-gap: 17px;

    .paginationButton {
      cursor: pointer;
      background: transparent;
      @include headingFour;
      color: $grey;
      transition: all .5s ease;

      &:hover {
        transform: scale(1.1);
        color: $darkFiolet;
      }
    }
  }
}
