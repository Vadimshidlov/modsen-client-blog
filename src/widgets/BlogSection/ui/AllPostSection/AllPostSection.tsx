import React from 'react';
import styles from '@/widgets/BlogSection/ui/AllPostSection/AllPostSection.module.scss';
import container from '@/shared/styles/container.module.scss';
import { POSTS } from '@/shared/constants/posts';
import { PostItem } from '@/widgets/PostItem';

export function AllPostSection() {
  return (
    <section className={`${styles.allPostSection} ${container.container}`}>
      <h1 className={styles.sectionTitle}>All Posts</h1>
      <div className={styles.postsContainer}>
        {POSTS.slice(0, 5).map(({ id, title, text, img, category }) => (
          <PostItem key={id} id={id} title={title} text={text} img={img} category={category} />
        ))}
      </div>
      <div className={styles.paginationController}>
        <button className={styles.paginationButton}>&lt; Prev</button>
        <button className={styles.paginationButton}>Next &gt;</button>
      </div>
    </section>
  );
}
