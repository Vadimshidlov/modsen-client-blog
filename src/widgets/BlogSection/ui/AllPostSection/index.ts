export { AllPostSection } from '@/widgets/BlogSection/ui/AllPostSection/AllPostSection';
