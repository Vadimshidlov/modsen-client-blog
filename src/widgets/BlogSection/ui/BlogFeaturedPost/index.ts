export { BlogFeaturedPost } from '@/widgets/BlogSection/ui/BlogFeaturedPost/BlogFeaturedPost';
