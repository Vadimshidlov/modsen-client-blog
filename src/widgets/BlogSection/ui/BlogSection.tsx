import React from 'react';
import { BlogFeaturedPost } from '@/widgets/BlogSection/ui/BlogFeaturedPost';
import { AllPostSection } from '@/widgets/BlogSection/ui/AllPostSection';

export function BlogSection() {
  return (
    <section>
      <BlogFeaturedPost />
      <AllPostSection />
    </section>
  );
}
