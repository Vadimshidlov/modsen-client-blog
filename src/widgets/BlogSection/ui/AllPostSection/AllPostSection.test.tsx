import React from 'react';
import { render } from '@testing-library/react';
import { AllPostSection } from '@/widgets/BlogSection/ui/AllPostSection';

jest.mock('next-intl', () => ({
  useTranslations: jest.fn().mockReturnValue((key: string) => `Mocked ${key}`),
}));

describe('AllPostSection tests', () => {
  test('should matches snapshot', () => {
    const { container } = render(<AllPostSection />);
    expect(container).toMatchSnapshot();
  });

  test('should renders pagination buttons correctly', () => {
    const { getByTestId } = render(<AllPostSection />);

    expect(getByTestId('prevButton')).toBeInTheDocument();
    expect(getByTestId('nextButton')).toBeInTheDocument();
  });

  test('should disables prev button on first page', () => {
    const { getByTestId } = render(<AllPostSection />);

    const prevButton = getByTestId('prevButton');
    expect(prevButton).toHaveAttribute('disabled');
  });

  test('should enables next button on first page', () => {
    const { getByTestId } = render(<AllPostSection />);

    const nextButton = getByTestId('nextButton');
    expect(nextButton).not.toHaveAttribute('disabled');
  });
});
