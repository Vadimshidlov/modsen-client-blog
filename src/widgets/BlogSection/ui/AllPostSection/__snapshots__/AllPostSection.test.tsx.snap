// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllPostSection tests should matches snapshot 1`] = `
<div>
  <section
    class="allPostSection container"
  >
    <h1
      class="sectionTitle"
    >
      Mocked title
    </h1>
    <div
      class="postsContainer"
    >
      <article
        class="postContainer"
      >
        <div
          class="postImageWrapper"
        >
          <img
            alt="post-image"
            class="postImage"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </div>
        <div
          class="postContentBlock"
        >
          <h4
            class="postSubTitle variable"
          >
            MOCKED TECHNOLOGY
          </h4>
          <a
            href="/posts/1"
          >
            <h2
              class="postTitle"
            >
              Step-by-step guide to choosing great font pairs
            </h2>
          </a>
          <span
            class="postText"
          >
            Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident.
          </span>
        </div>
      </article>
      <article
        class="postContainer"
      >
        <div
          class="postImageWrapper"
        >
          <img
            alt="post-image"
            class="postImage"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </div>
        <div
          class="postContentBlock"
        >
          <h4
            class="postSubTitle variable"
          >
            MOCKED STARTUP
          </h4>
          <a
            href="/posts/2"
          >
            <h2
              class="postTitle"
            >
              Design tips for designers that cover everything you need
            </h2>
          </a>
          <span
            class="postText"
          >
            Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident.
          </span>
        </div>
      </article>
      <article
        class="postContainer"
      >
        <div
          class="postImageWrapper"
        >
          <img
            alt="post-image"
            class="postImage"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </div>
        <div
          class="postContentBlock"
        >
          <h4
            class="postSubTitle variable"
          >
            MOCKED BUSINESS
          </h4>
          <a
            href="/posts/3"
          >
            <h2
              class="postTitle"
            >
              How to build rapport with your web design clients
            </h2>
          </a>
          <span
            class="postText"
          >
            Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident.
          </span>
        </div>
      </article>
      <article
        class="postContainer"
      >
        <div
          class="postImageWrapper"
        >
          <img
            alt="post-image"
            class="postImage"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </div>
        <div
          class="postContentBlock"
        >
          <h4
            class="postSubTitle variable"
          >
            MOCKED STARTUP
          </h4>
          <a
            href="/posts/4"
          >
            <h2
              class="postTitle"
            >
              Logo design trends to avoid in 2022
            </h2>
          </a>
          <span
            class="postText"
          >
            Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident.
          </span>
        </div>
      </article>
      <article
        class="postContainer"
      >
        <div
          class="postImageWrapper"
        >
          <img
            alt="post-image"
            class="postImage"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </div>
        <div
          class="postContentBlock"
        >
          <h4
            class="postSubTitle variable"
          >
            MOCKED TECHNOLOGY
          </h4>
          <a
            href="/posts/5"
          >
            <h2
              class="postTitle"
            >
              8 Figma design systems you can download for free today
            </h2>
          </a>
          <span
            class="postText"
          >
            Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident.
          </span>
        </div>
      </article>
    </div>
    <div
      class="paginationController"
    >
      <button
        class="paginationButton"
        data-testid="prevButton"
        disabled=""
      >
        &lt; 
        Mocked prevButtonText
      </button>
      <button
        class="paginationButton"
        data-testid="nextButton"
      >
        Mocked nextButtonText
         &gt;
      </button>
    </div>
  </section>
</div>
`;
